presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/confluent/ConfluentSchemaRegistryTableDescriptionSupplier.java

```diff
@@ -0,0 +1,367 @@
+            this.schemaRegistryClient = requireNonNull(schemaRegistryClient, "schemaRegistryClient is null");
+            this.defaultSchema = kafkaConfig.getDefaultSchema();
+            this.subjectsCacheRefreshInterval = confluentConfig.getConfluentSubjectsCacheRefreshInterval();
+            this.typeManager = typeManager;
+        }
+
+        @Override
+        public TableDescriptionSupplier get()
+        {
+            return new ConfluentSchemaRegistryTableDescriptionSupplier(schemaRegistryClient, defaultSchema, subjectsCacheRefreshInterval, typeManager);
+        }
+    }
+
+    // Refresh mapping of topic to subjects.
+    // Note that this mapping only supports subjects that use the SubjectNameStrategy of TopicName.
+    // If another SubjectNameStrategy is used then the key and message subject must be encoded
+    // into the table name as follows:
+    // <table name>[&key-subject=<key subject>][&message-subject=<message subject]
+    // ex. kafka.default."my-topic&key-subject=foo&message-subject=bar"
+    private Map<String, TopicAndSubjects> refreshSubjects()
+    {
+        try {
+            Map<String, Set<String>> topicToSubjects = new HashMap<>();
+            for (String subject : schemaRegistryClient.getAllSubjects()) {
+                if (isValidSubject(subject)) {
+                    topicToSubjects.computeIfAbsent(extractTopicFromSubject(subject), k -> new HashSet<>()).add(subject);
+                }
+            }
+            ImmutableMap.Builder<String, TopicAndSubjects> topicSubjectsCacheBuilder = ImmutableMap.builder();
+            for (Map.Entry<String, Set<String>> entry : topicToSubjects.entrySet()) {
+                String topic = entry.getKey();
+                TopicAndSubjects topicAndSubjects = new TopicAndSubjects(topic,
+                        getKeySubjectFromTopic(topic, entry.getValue()),
+                        getValueSubjectFromTopic(topic, entry.getValue()));
+                topicSubjectsCacheBuilder.put(topicAndSubjects.getTableName(), topicAndSubjects);
+            }
+            return topicSubjectsCacheBuilder.build();
+        }
+        catch (IOException | RestClientException e) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, "Failed to retrieve subjects from schema registry", e);
+        }
+    }
+
+    @Override
+    public Optional<KafkaTopicDescription> getTopicDescription(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        requireNonNull(schemaTableName, "schemaTableName is null");
+        TopicAndSubjects topicAndSubjects = parseTopicAndSubjects(schemaTableName);
+
+        String tableName = topicAndSubjects.getTableName();
+        String topic = topicAndSubjects.getTopic();
+        Optional<String> keySubject = topicAndSubjects.getKeySubject();
+        Optional<String> valueSubject = topicAndSubjects.getValueSubject();
+
+        TopicAndSubjects topicAndSubjectsFromCache = topicAndSubjectsSupplier.get().get(tableName);
```

Comment:
Move this up, then you can:

```
        Optional<String> keySubject = topicAndSubjects.getKeySubject()
                .or(topicAndSubjectsFromCache::getKeySubject);
```
Same couldbe applied to value. Mind that `topicAndSubjectsFromCache` is nullable.

Reply:
The issue is when the topic is mixed case (there are tests for this). If the topic is encoded in the table name, it will be lowercase.